Source/WebCore/dom/Position.cpp

```diff
@@ -696,40 +704,41 @@ Position Position::upstream(EditingBoundaryCrossingRule rule) const
         return { };
     
     // iterate backward from there, looking for a qualified position
-    Node* boundary = enclosingVisualBoundary(startNode.get());
+    RefPtr boundary = enclosingVisualBoundary(startNode.get());
     // FIXME: PositionIterator should respect Before and After positions.
-    PositionIterator lastVisible = m_anchorType == PositionIsAfterAnchor ? makeDeprecatedLegacyPosition(m_anchorNode.get(), caretMaxOffset(*m_anchorNode)) : *this;
+    auto anchorNode = protectedAnchorNode();
+    PositionIterator lastVisible = m_anchorType == PositionIsAfterAnchor ? makeDeprecatedLegacyPosition(anchorNode.get(), caretMaxOffset(*anchorNode)) : *this;
     PositionIterator currentPosition = lastVisible;
     bool startEditable = startNode->hasEditableStyle();
     RefPtr lastNode = startNode;
     bool boundaryCrossed = false;
     for (; !currentPosition.atStart(); currentPosition.decrement()) {
-        auto& currentNode = *currentPosition.node();
+        RefPtr currentNode = currentPosition.node();
```

Comment:
Why aren't we using a `Ref` here to be closer to the old code and make it clear it can't be null?

Reply:
Sure, changed.